javascript/javascript3/week3/shoppingcart.js

```diff
@@ -0,0 +1,98 @@
+class Product {
+    constructor(name, price) {
+        this.name = name;
+        this.price = price;
+    }
+    convertToCurrency(currency) {
+        fetch(
+                "https://v6.exchangerate-api.com/v6/2094ae77092b17e58dd73819/latest/DKK"
+            )
+            .then((response) => response.json())
+            .then((data) => {
+                const currencies = data.conversion_rates;
+                const convertCurrency = this.price * currencies[currency];
+                console.log(convertCurrency);
+            });
+    }
+}
+
+class ShoppingCart {
+  constructor() {
+    this.products = [];
+  }
+
+  addProduct(product) {
+    this.products.push(product);
+    console.log(this.products);
+  }
+
+  removeProduct(product) {
+    this.products = this.products.filter((item) => item !== product);
+    console.log(this.products);
+  }
+
+  searchProduct(productName) {
+    const searchedProduct = this.products.filter(
+      (product) => product.name === productName
+    );
+    console.log(searchedProduct);
+  }
+
+  getTotal() {
+    let total = 0;
+    const totalPrice = this.products.reduce((accumulator, currentValue) => {
+      return accumulator + currentValue.price;
+    }, total);
+    console.log(`The total price is : ${totalPrice}`);
```

Comment:
usually better to return the value, and do a console log _outside_ the function.